unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/net/DatagramSocketTest.scala

```diff
@@ -0,0 +1,308 @@
+package org.scalanative.testsuite.javalib.net
+
+import java.net._
+
+import org.junit.Test
+import org.junit.Assert._
+import org.junit.Assume._
+
+import java.io.IOException
+
+import org.scalanative.testsuite.utils.AssertThrows.assertThrows
+
+class DatagramSocketTest {
+  @Test def constructor(): Unit = {
+    val ds = new DatagramSocket(0, InetAddress.getLocalHost())
+    try {
+      assertTrue("Created socket with incorrect port", ds.getLocalPort() != 0)
+      assertEquals(
+        "Created socket with incorrect address",
+        InetAddress.getLocalHost(),
+        ds.getLocalAddress()
+      )
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def close(): Unit = {
+    val ds = new DatagramSocket(0)
+    ds.close()
+    val dp = new DatagramPacket(
+      "Test String".getBytes,
+      11,
+      InetAddress.getLocalHost,
+      0
+    )
+    assertThrows(classOf[IOException], ds.send(dp))
+  }
+
+  @Test def getLocalPort(): Unit = {
+    val ds = new DatagramSocket()
+    try {
+      assertTrue("Returned incorrect port", ds.getLocalPort != 0)
+    } finally {
+      ds.close()
+    }
+  }
+
+  @Test def getPort(): Unit = {
+    val ds = new DatagramSocket()
+    try {
+      assertEquals(
+        "Expected -1 for remote port as not connected",
```

Comment:
I think this needs wordsmithing/bike-shedding.
Something like "Unexpected remote port" with a comment
on the -1 line that Datagram Socket is not connected at this point.